PcsConfig/DeviceGroupsTests.cs

```diff
@@ -0,0 +1,280 @@
+            };
+            var request = new HttpRequest();
+            request.SetContent(inputDeviceGroup);
+            request.SetUriFromString(Constants.CONFIG_ADDRESS + "/devicegroups");
+            request.Options.Timeout = TIMEOUT;
+
+            // Act for create
+            var response = this.httpClient.PostAsync(request).Result;
+
+            // Assert for create
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var createdDeviceGroup = JsonConvert.DeserializeObject<DeviceGroupApiModel>(response.Content);
+            Assert.Equal(inputDeviceGroup.DisplayName, createdDeviceGroup.DisplayName);
+            Assert.Equal(inputDeviceGroup.Conditions.ToString(), createdDeviceGroup.Conditions.ToString());
+
+            // Arange for get
+            request.SetUriFromString(Constants.CONFIG_ADDRESS + "/devicegroups/" + createdDeviceGroup.Id);
+            request.SetContent("");
+
+            // Act for get
+            response = this.httpClient.GetAsync(request).Result;
+
+            // Assert for get
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            createdDeviceGroup = JsonConvert.DeserializeObject<DeviceGroupApiModel>(response.Content);
+            Assert.Equal(inputDeviceGroup.DisplayName, createdDeviceGroup.DisplayName);
+            Assert.Equal(inputDeviceGroup.Conditions.ToString(), createdDeviceGroup.Conditions.ToString());
+        }
+
+        /// <summary>
+        /// Test that the service can get a created Device Group
+        /// </summary>
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void ShouldGetMultipleCreatedDeviceGroup()
+        {
+            // Arange for create 1
+            var inputDeviceGroup = new DeviceGroupApiModel
+            {
+                Id = "",
+                DisplayName = "TEST-CreateDeviceGroup-1",
+                Conditions = CreateFakeDeviceGroupConditionList(),
+                ETag = "",
+                Metadata = null
+            };
+            var request = new HttpRequest();
+            request.SetContent(inputDeviceGroup);
+            request.SetUriFromString(Constants.CONFIG_ADDRESS + "/devicegroups");
+            request.Options.Timeout = TIMEOUT;
+
+            // Act for create 1
+            var response = this.httpClient.PostAsync(request).Result;
+
+            // Assert for create 1
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var createdDeviceGroup = JsonConvert.DeserializeObject<DeviceGroupApiModel>(response.Content);
+            Assert.Equal(inputDeviceGroup.DisplayName, createdDeviceGroup.DisplayName);
+            Assert.Equal(inputDeviceGroup.Conditions.ToString(), createdDeviceGroup.Conditions.ToString());
+
+            // Dispose
+            disposeList.Add(createdDeviceGroup.Id);
+
+            // Arange for create 2
+            inputDeviceGroup.DisplayName = "TEST - CreateDeviceGroup - 2";
```

Comment:
When you create a private method that creates device group pass in DisplayName as a parameter so you can do multiple creates as well

Reply:
done